src/main/java/edu/cornell/kfs/coa/document/validation/impl/GlobalIndirectCostRecoveryAccountsRule.java

```diff
@@ -183,208 +169,193 @@ protected boolean checkIndirectCostRecoveryAccount(IndirectCostRecoveryAccountCh
+					result = false;
+				}
+			}
 		}
-		
-		for (GlobalBusinessObjectDetailBase globalDetail : globalDetails) {
-			List<IndirectCostRecoveryAccount> existingIcrAccounts = new ArrayList<IndirectCostRecoveryAccount>();
 
-			SubAccountGlobalDetail subAccountGlobalDetail = (SubAccountGlobalDetail) globalDetail;
+		return result;
+	}
+	
 
-			subAccountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.SUB_ACCOUNT);
-			
-			List<A21IndirectCostRecoveryAccount> a21IcrAccounts = subAccountGlobalDetail.getSubAccount().getA21SubAccount().getA21IndirectCostRecoveryAccounts();
-			List<IndirectCostRecoveryAccount> currentActiveIndirectCostRecoveryAccountList = new ArrayList<IndirectCostRecoveryAccount>();
-			List<IndirectCostRecoveryAccount> existingICRs = new ArrayList<IndirectCostRecoveryAccount>(); 
-			List<IndirectCostRecoveryAccount> existingICRsWorkList = new ArrayList<IndirectCostRecoveryAccount>();
-			
-			for (A21IndirectCostRecoveryAccount a21ICRAccount : a21IcrAccounts) {
-				IndirectCostRecoveryAccount icrAcct = new IndirectCostRecoveryAccount();
-				icrAcct.setChartOfAccountsCode(a21ICRAccount.getChartOfAccountsCode());
-				icrAcct.setAccountNumber(a21ICRAccount.getAccountNumber());
-				icrAcct.setIndirectCostRecoveryAccountNumber(a21ICRAccount.getIndirectCostRecoveryAccountNumber());
-				icrAcct.setIndirectCostRecoveryFinCoaCode(a21ICRAccount.getIndirectCostRecoveryFinCoaCode());
-				icrAcct.setAccountLinePercent(a21ICRAccount.getAccountLinePercent());
-				icrAcct.setActive(a21ICRAccount.isActive());
-				existingIcrAccounts.add(icrAcct);
+	protected boolean checkICRAccuntTotalDistributionOnDetailWillBe100PercentAfterUpdate(List<IndirectCostRecoveryAccountChange> icrUpdates, List<IndirectCostRecoveryAccount> existingICRs, GlobalBusinessObjectDetailBase globalDetail, GlobalObjectWithIndirectCostRecoveryAccounts globalObjectWithIndirectCostRecoveryAccounts) {
+		List<IndirectCostRecoveryAccount> currentActiveIndirectCostRecoveryAccountList = buildCurrentActiveICRAccountsAfterICRUpdates(icrUpdates, existingICRs, globalDetail, globalObjectWithIndirectCostRecoveryAccounts);
+
+		if ((ObjectUtils.isNull(currentActiveIndirectCostRecoveryAccountList) || (currentActiveIndirectCostRecoveryAccountList.size() == 0))) {
+			return true;
+		} else {
+
+			if (totalDistributionOnUpdatedActiveICRsIs100Percent(currentActiveIndirectCostRecoveryAccountList)) {
+				return true;
+			} else {
+				return false;
 			}
+		}
+	}
+	
 
-			for (IndirectCostRecoveryAccount icrAccount : existingIcrAccounts) {
-				existingICRsWorkList.add(icrAccount);
+	protected boolean totalDistributionOnUpdatedActiveICRsIs100Percent(List<IndirectCostRecoveryAccount> currentActiveIndirectCostRecoveryAccountList) {
+		BigDecimal totalDistribution =  computeTotalDistribution(currentActiveIndirectCostRecoveryAccountList);
+		if (totalDistribution.compareTo(ONE_HUNDRED_PERCENT) == 0) {
+			return true;
+		}
+		else{
+			return false;
+		}
```

Comment:
Another area where the "if" check result could be returned directly instead of using the "return true/false" branching.